src/category_theory/preadditive/injective.lean

```diff
@@ -301,5 +301,68 @@ by convert congr_arg quiver.hom.unop
 end
 
 end injective
+namespace adjunction
 
+variables {D : Type*} [category D] {F : C ⥤ D} {G : D ⥤ C}
+
+lemma map_injective (adj : F ⊣ G) [F.preserves_monomorphisms] (I : D) (hI : injective I) :
+  injective (G.obj I) :=
+begin
+  constructor,
+  intros X Y f g hg,
+  haveI : mono (F.map g) := by unfreezingI { apply_instance },
+  rcases @hI.1 (F.map f ≫ adj.counit.app _) (F.map g),
+  use adj.unit.app Y ≫ G.map w,
+  rw [←unit_naturality_assoc, ←G.map_comp, h],
+  simp,
+end
```

Comment:
Using `introI`, the `mono` instance is obtained automatically. Also, `hI.factors` is more friendly to the reader than `@hI.1`. (The same comment also applies to `injective_of_map_injective`, `map_projective` and `projective_of_map_projective`.)
```suggestion
⟨λ X Y f g, begin
  introI,
  rcases hI.factors (F.map f ≫ adj.counit.app _) (F.map g),
  use adj.unit.app Y ≫ G.map w,
  rw [←unit_naturality_assoc, ←G.map_comp, h],
  simp,
end⟩
```